Dockerfile

```diff
@@ -0,0 +1,47 @@
+# syntax=docker/dockerfile:1
+FROM node:16 as base
+
+RUN apt update && apt install -y \
+        apt-transport-https \
+        bash \
+        ca-certificates \
+        curl \
+        git \
+        gnupg2 \
+        lsb-release \
+        python3 \
+        software-properties-common
+
+# Install docker
+RUN curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add -
+RUN add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/debian buster stable"
+RUN apt update && apt install -y \
+        containerd.io \
+        docker-ce \
+        docker-ce-cli \
+        docker-compose-plugin \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install kubectl cli
+RUN curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+RUN chmod +x ./kubectl
+RUN mv ./kubectl /usr/local/bin/kubectl
+
+# Install greenframe cli
+RUN cd /root \
+        && curl https://assets.greenframe.io/install.sh | bash
+ENV PATH $PATH:/root/.local/bin
+
+RUN mkdir /app
+RUN git config --global --add safe.directory /app
+WORKDIR /app
+
+
+## Specific build with GCP tools
+FROM base as gcp
```

Comment:
How is this part related to greenframe?

Reply:
`gcloud` command line tools are required to use `kubectl ...` commands in the Google Cloud Platform kubernetes execution context, especially for all authentication steps.
It's the same for Azure and AWS.

I think it would be great to offer all the necessary tools for the major cloud providers (GCP, AWS, Azure) and create a specific version of the greenframe-cli image.

I added a new make command (which can be used in the GitHub workflow) to build these images:
```
DOCKER_BUILDKIT=1 docker build -t greenframe-cli:latest -t greenframe-cli:$(PACKAGE_VERSION) . --target base
```
for the base image, and:
```
DOCKER_BUILDKIT=1 docker build -t greenframe-cli:$(PACKAGE_VERSION)-gcp . --target gcp 
etc ...
``` 
for the specific cloud providers

One greenframe-cli version will have these docker version:
* greenframe-cli:latest
* greenframe-cli:1.4.4
* greenframe-cli:1.4.4-gcp
* greenframe-cli:1.4.4-azure
* greenframe-cli:1.4.4-...

Reply:
This is not something we want to maintain in this repository. I prefer that we provide a cloud-agnostic base image, and leave to the community the cloud-specific images.